datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaTopicPartitionTracker.java

```diff
@@ -75,22 +76,30 @@ public void onPartitionsRevoked(@NotNull Collection<TopicPartition> topicPartiti
       }
     });
 
-    // Remove consumer offsets for partitions that have been revoked. The reason to remove the consumer offsets
+    // Remove consumed offsets for partitions that have been revoked. The reason to remove the consumed offsets
     // here is that another host may handle these partitions due to rebalance, and we don't want to have duplicate
     // consumer offsets for affected partitions (even though the ones with larger offsets wins).
+    removeOffsetsForTopicPartition(topicPartitions, _consumedOffsets);
+
+    // Remove committed offsets for partitions that have been revoked.
+    removeOffsetsForTopicPartition(topicPartitions, _committedOffsets);
+  }
+
+  private void removeOffsetsForTopicPartition(@NotNull Collection<TopicPartition> topicPartitions,
+      Map<String, Map<Integer, Long>> committedOffsets) {
```

Comment:
Nitpick. May be you need to call this just offsets since its being used by both _consumedOffsets and _committedOffsets

Reply:
+1 make this generic. We also intend to add end offsets to this list.